src/duckling/shell/shell.component.ts

```diff
@@ -35,23 +41,73 @@ import {WindowService} from '../util';
         </div>
     `
 })
-export class ShellComponent {
+export class ShellComponent implements OnDestroy {
+    private _imagesLoaded : boolean = false;
+    private _numImagesToLoad = 0;
```

Comment:
This logic doesn't make sense to in the shell. The asset service should create a promise or an observable that fires once all requested assets have been loaded.